staging/src/k8s.io/externaljwt/util.go

```diff
@@ -0,0 +1,53 @@
+package externaljwt
+
+import (
+	"fmt"
+
+	jose "gopkg.in/square/go-jose.v2"
+	"gopkg.in/square/go-jose.v2/jwt"
+)
+
+type PublicKey struct {
+	KeyID                    string
+	PublicKey                interface{}
+	ExcludeFromOIDCDiscovery bool
+}
+
+// Listener is an interface to use to notify interested parties of a change.
+type Listener interface {
+	// Enqueue should be called when an input may have changed
+	Enqueue()
+}
+
+// GenerateToken is shared between internal and external signer code.
+func GenerateToken(signer jose.Signer, iss string, claims *jwt.Claims, privateClaims interface{}) (string, error) {
+	// claims are applied in reverse precedence
+	return jwt.Signed(signer).
+		Claims(privateClaims).
+		Claims(claims).
+		Claims(&jwt.Claims{
+			Issuer: iss,
+		}).
+		CompactSerialize()
+}
+
+func MergeClaims(iss string, claims *jwt.Claims, privateClaims interface{}) ([]byte, error) {
+	var out []byte
+	signer := payloadGrabber(func(payload []byte) { out = payload })
+	_, err := GenerateToken(signer, iss, claims, privateClaims)
+	if len(out) == 0 {
+		return nil, fmt.Errorf("failed to marshal: %w", err)
+	}
+	return out, nil // error is safe to ignore as long as we have the payload bytes
+}
+
+var _ jose.Signer = payloadGrabber(nil)
+
+type payloadGrabber func(payload []byte)
+
+func (p payloadGrabber) Sign(payload []byte) (*jose.JSONWebSignature, error) {
+	p(payload)
+	return nil, jose.ErrUnprotectedNonce // return some error to stop after we have the payload
+}
+
+func (p payloadGrabber) Options() jose.SignerOptions { return jose.SignerOptions{} }
```

Comment:
Unexport `MergeClaims` and move these lines to `plugin.go`.  They aren't shared utils, but instead are specific to the external signer plugin itself.

Reply:
ohh, we are using it from jwt.go now.

Reply:
Only `GenerateToken` is used outside of this package as an util, the rest are specific to `plugin.go`.

Reply:
Done